### PR TITLE
Adds Spray Bottles

### PR DESCRIPTION
**Pull request by exp111**
Pls no Merge Conflicts

### CODE REVIEW

**Comment by exp111** [comment]
The original spray bottle applies vapor, not instantly create a puddle. 
Should this maybe wait for atmos?

**Comment by Zumorica** [comment]
> The original spray bottle applies vapor, not instantly create a puddle.
> Should this maybe wait for atmos?

Not really. This should create some sort of "vapor" entity that moves transferring reagents to the ground puddles until it runs out of reagents.

File: Content.Server/GameObjects/Components/Chemistry/VaporHelper.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        /// <summary>
+        /// Spawns the specified solution as a vapor at the entity's location if possible.
+        /// </summary>
+        /// <param name="entity">Entity to spawn the vapor at</param>
+        /// <param name="solution">Initial solution for the prototype</param>
+        internal static void SpillAt(IEntity entity, Solution solution)
+        {
+            var entityLocation = entity.Transform.GridPosition;
+            SpillAt(entityLocation, solution);
```
Comment (Zumorica) [inline Content.Server/GameObjects/Components/Chemistry/VaporHelper.cs:23]: Return VaporComponent here, perhaps?

Comment (exp111) [reply Content.Server/GameObjects/Components/Chemistry/VaporHelper.cs:23]: Literally stole that from SpillHelper. Should I add it there too?

**Comment by exp111** [comment]
The vapor currently doesn't move. I thought the vapor just is a delayed puddle spawner.

**Comment by Zumorica** [comment]
As far as I remember, on TG (and some others) the vapor moves in the direction you clicked, cleaning the tiles it touches and leaving puddles behind

**Comment by exp111** [comment]
[code](https://github.com/tgstation/tgstation/blob/732d83f249a3a6f60b1e31f9fcea6a6561d90dd3/code/modules/reagents/reagent_containers/spray.dm#L80)
if i'm not dumb rn, it goes through all tiles in the line and let's the tile react
i'm doing this but spawning a vapor entity

you could also spawn a vapor entity and let that do the moving stuff. pretty much is the same result

**Comment by RemieRichards** [comment]
It uses the reagents from the "chempuff" entity.

**Comment by exp111** [comment]
well that's retarded

**Comment by RemieRichards** [comment]
🤷‍♀️

**Comment by Zumorica** [comment]
image
fix this

**Comment by exp111** [comment]
> fix this

Delete this

**Pull request merged**